compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -421,10 +421,12 @@ class AMDAIEUnrollLocalLoops : public OpRewritePattern<scf::ForOp> {
         return WalkResult::advance();
       }
 
-      uint64_t sourceMemspace =
-          dmaOp.getSourceObjectFifo().getMemorySpaceAsUInt();
-      uint64_t targetMemspace =
-          dmaOp.getTargetObjectFifo().getMemorySpaceAsUInt();
+      uint8_t sourceMemspace =
+          cast<AMDAIE::LogicalObjectFifoType>(dmaOp.getSource().getType())
+              .getMemorySpaceAsUInt();
+      uint8_t targetMemspace =
+          cast<AMDAIE::LogicalObjectFifoType>(dmaOp.getTarget().getType())
+              .getMemorySpaceAsUInt();
```

Comment:
It'd perhaps be nice to add this as an API in `AMDAIE_DmaCpyNdBaseOp` itself?

Reply:
Done